ludwig/models/llm.py

```diff
@@ -142,6 +143,11 @@ def to_device(self, device):
                 self.model.save_pretrained(tmpdir)
                 self.model = AutoModelForCausalLM.from_pretrained(tmpdir, **model_kwargs)
 
+            # If loading from checkpoint did not place it on the desired device, forcefully place it on
+            # the desired device. On Cuda, this will place it on the device with the lowest index.
+            if self.model.device != device:
```

Comment:
What would cause this to happen? Line above should handle this. right? My concern would be what would happen if the model was too big for one gpu and we tried to do this.

Reply:
I'm not 100% sure, but it happens specifically when using the local backend on a pod with GPUs

Reply:
Actually I agree, this can probably go. It's only needed for models where device_map isn't implemented, in which case we'd want to get rid of it from `model_kwargs` to begin wtih